packages/serverpod_client/lib/src/connectivity_monitor.dart

```diff
@@ -25,12 +25,18 @@ abstract class ConnectivityMonitor {
     _listeners.clear();
   }
 
+  /// Only notify listeners if the connectivity status has changed.
+  bool _prevConnected = false;
```

Comment:
Suggestion: make this nullable as we otherwise will not notify anyone if the first event is going from connected to disconnected.